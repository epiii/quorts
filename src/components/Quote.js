import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import IconClose from "react-icons/lib/md/close";
import IconCopy from "react-icons/lib/md/content-copy";

import "typeface-roboto";

const Container = styled.div`
  padding: 16px;
  padding-top: 32px;
  box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.3);
  position: relative;
`;

const Text = styled.p`
  margin-top: 0px;
  margin-bottom: 24px;
  font-family: Roboto;
  font-size: 16px;
  color: #333;
  line-height: 1.5em;
`;

export const Button = styled.span`
  margin: 0px;
  font-family: Roboto;
  font-size: 16px;
  color: #aaa;
  cursor: pointer;
  user-select: none;
`;

export const StyledClose = styled(IconClose)`
  color: #aaa;
  font-size: 20px;
  padding: 4px;
  position: absolute;
  top: 4px;
  right: 4px;
  cursor: pointer;
`;

const StyledCopy = styled(IconCopy)`
  font-size: 20px;
`;

const Quote = ({ text, onClickCopy, onClickRemove }) => (
  <Container>
    <StyledClose onClick={onClickRemove} />
    <Text>{text}</Text>
    <Button onClick={onClickCopy}>
      <StyledCopy /> COPY
    </Button>
  </Container>
);

Quote.propTypes = {
  text: PropTypes.string,
  onClickCopy: PropTypes.func,
  onClickRemove: PropTypes.func
};

export default Quote;
