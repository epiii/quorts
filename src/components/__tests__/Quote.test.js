import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Quote, { Button, StyledClose } from "../Quote";

describe("Quote", () => {
  const setup = propOverrides => {
    const props = {
      text: "I am React Developer",
      onClickRemove: jest.fn(),
      onClickCopy: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Quote {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should calls 'onClickCopy' while Copy button was clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find(Button).simulate("click");
    expect(props.onClickCopy).toBeCalled();
  });

  it("should calls 'onClickRemove' while Remove button was clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find(StyledClose).simulate("click");
    expect(props.onClickRemove).toBeCalled();
  });
});
