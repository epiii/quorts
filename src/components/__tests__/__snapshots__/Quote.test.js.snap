// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should renders default 1`] = `
<styled.div>
  <Styled(MdClose)
    onClick={[Function]}
  />
  <styled.p>
    I am React Developer
  </styled.p>
  <styled.span
    onClick={[Function]}
  >
    <Styled(MdContentCopy) />
     COPY
  </styled.span>
</styled.div>
`;
